### PR TITLE
New 1055 searchtag variations stvi

### PR DESCRIPTION
**Pull request by stanvins**

### CODE REVIEW

File: e2e/cypress/integration/search.spec.js [hunk 1]
```diff
@@ -17,4 +17,116 @@ describe('Search test', function() {
...
+    cy.get(
+      ':nth-child(1) > [data-cy=workcard-1] > .work-card__content > [data-cy=book-cover-loaded] > .hover-details-fade'
+    ).click();
+    cy.get('[data-cy=pages-count]')
+      .invoke('text')
+      .then(value => {
+        if (value === '') {
+          return;
+        }
+        expect(value).to.be.greaterThan(350);
```
Comment (kfudbc) [inline e2e/cypress/integration/search.spec.js]: Jeg er ikke sikker på om du tester at tagget ændret udseende til et "must-have" tag?

File: src/client/components/filter/SearchBar/SearchBar.component.js [hunk 2]
```diff
@@ -67,15 +82,37 @@ const SelectedQuery = ({selected, onRemove}) => (
...
   </Button>
 );
+
 class SearchBar extends React.Component {
+  constructor() {
+    super();
+    this.state = {query: '', expanded: false};
+  }
+
+  getReqState = s => {
```
Comment (kfudbc) [inline src/client/components/filter/SearchBar/SearchBar.component.js]: Skulle denne evt eksporteres fra withTagsFromUrl?

File: e2e/cypress/integration/search.spec.js [hunk 3]
```diff
@@ -17,4 +17,116 @@ describe('Search test', function() {
...
+      .type(searchWord1);
+
+    cy.wait(1000);
+
+    cy.get('[data-cy=suggestion-element]')
+      .contains('Lang')
+      .first()
+      .click();
+    cy.get(
+      '.topbar__search-bar--wrap > #selected-filters-wrap > #selectedFilters > [data-cy] > :nth-child(1) > span'
```
[the remark this answers is not in the document]
Comment (stanvins) [reply e2e/cypress/integration/search.spec.js]: In this test, I want to make sure a must have tag - in this case "lang" returns books with more than 350 pages. Its not a particular book I am interested in.

File: e2e/cypress/integration/search.spec.js (hunk 1 above)
Comment (stanvins) [reply e2e/cypress/integration/search.spec.js]: Ah. I see. Maybe the name of the test is wrong. I am not checking if the tag is underlined, only that the function I created only lets the right pids through.

File: src/client/components/filter/SearchBar/SearchBar.component.js (hunk 2 above)
Comment (stanvins) [reply src/client/components/filter/SearchBar/SearchBar.component.js]: ... moved the getReqState function into withTagsFromUrl

File: e2e/cypress/integration/search.spec.js (hunk 1 above)
Comment (stanvins) [reply e2e/cypress/integration/search.spec.js]: Made the test more robust, checking both that the tags are formatted properly and that it returns what is expected.

**Pull request merged**